lib/manageiq/api/client/connection.rb

```diff
@@ -0,0 +1,87 @@
+module ManageIQ
+  module API
+    module Client
+      class Connection
+        attr_accessor :url
+        attr_accessor :authentication
+        attr_accessor :response
+        attr_accessor :error
+
+        API_PREFIX = "/api".freeze
+        CONTENT_TYPE = "application/json".freeze
+
+        def initialize(url, authentication)
+          @url = url
+          @authentication = authentication
+          @error = ManageIQ::API::Client::Error.new
```

Comment:
I don't like creating an error object on every initialize.  There might be a better way to do it at the moment it's called if you need it in non-error cases.